src/frontend/src/storage/index.test.ts

```diff
@@ -146,7 +146,46 @@ test(
 );
 
 test(
-  "anchors are also written to V2",
+  "V3 anchors are migrated",
+  withStorage(
+    async () => {
+      expect(await getAnchors()).toContain(BigInt(10000));
+      expect(await getAnchors()).toContain(BigInt(10001));
+      expect(await getAnchors()).toContain(BigInt(10003));
+    },
+    {
+      indexeddb: {
+        before: {
+          /* V3 layout */
+          "ii-storage-v3": {
+            anchors: {
+              "10000": { lastUsedTimestamp: 0, knownPrincipals: [] },
+              "10001": { lastUsedTimestamp: 0, knownPrincipals: [] },
+              "10003": { lastUsedTimestamp: 0, knownPrincipals: [] },
+            },
+            hasher: await crypto.subtle.generateKey(
+              { name: "HMAC", hash: "SHA-512" },
+              false /* not extractable */,
+              ["sign"] /* only used to "sign" (e.g. produce a digest ) */
+            ),
+          },
+        },
+        after: (storage) => {
+          // Written to V3
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const storageV4: any = storage["ii-storage-v4"];
+          expect(storageV4).toBeTypeOf("object");
+          expect(storageV4.anchors["10000"]).toBeDefined();
+          expect(storageV4.anchors["10001"]).toBeDefined();
+          expect(storageV4.anchors["10003"]).toBeDefined();
```

Comment:
Could you also check here that `getAnchorByPrincipal` works as expected?

Reply:
No, known principals are _not_ migrated. They cant be migrated, because the digest scheme has changed.

Reply:
Using `getAnchorByPrincipal` on `10003` won't work in this scenario?

Reply:
No, it will return undefined. But the VC flow is currently the only usage of that (and it will be restored after the next login) so the impact for users should be acceptable.

Reply:
Ok, understood. Thanks for the explanation.